File: 1/1.3/Go/1.3.go
package main

import (
	"fmt"
	"strings"
)

func URIWhitespace(s string) string {
	return strings.ReplaceAll(strings.TrimSpace(s), " ", "%20")
}

func main() {
	fmt.Println(URIWhitespace("NoSpace"))
	fmt.Println(URIWhitespace("With Space"))
	fmt.Println(URIWhitespace("Many Spaces In Line"))
	fmt.Println(URIWhitespace(" StartWithSpace"))
	fmt.Println(URIWhitespace("EndWithSpace "))
	fmt.Println(URIWhitespace(" Start With Space And Multiple Spaces"))
	fmt.Println(URIWhitespace("End With Space And Multiple Spaces "))
	fmt.Println(URIWhitespace(" All In Spaces "))
}

File: 1/1.3/Python/1.3.py
def URIWhitespace(s: str) -> str:
    return s.strip().replace(" ", "%20")


if __name__ == '__main__':
    print(URIWhitespace("NoSpace"))
    print(URIWhitespace("With Space"))
    print(URIWhitespace("Many Spaces In Line"))
    print(URIWhitespace(" StartWithSpace"))
    print(URIWhitespace("EndWithSpace "))
    print(URIWhitespace(" Start With Space And Multiple Spaces"))
    print(URIWhitespace("End With Space And Multiple Spaces "))
    print(URIWhitespace(" All In Spaces "))
